src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+public class Parser {
+
+    public static Task parseInput(String input) {
+        if (input.startsWith("todo")) {
+            String substr = input.replaceFirst("todo", "").stripLeading();
+            return new ToDo(substr);
+            // System.out.println(taskList.addToDo(substr));
+        } else if (input.startsWith("deadline")) {
+            String substr = input.replaceFirst("deadline", "").stripLeading();
+            String[] substrArray = substr.split(" /by ", 2);
```

Comment:
Perhaps substrings is a better name than substrArray? Using plural form will distinguish clearly between single-valued and multi-valued variables.

Reply:
Nice catch on this, I did this before I read the guidelines and did tutorial 1, and took my existing code for granted.